CMake: Add small lib selection by default for baremetal profile

rajkan01 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->

- Select a small C library in the build when baremetal profile chosen.
- Replace CMake PUBLIC with INTERFACE keyword for all target_compile_options and target_link_options for mbed-core library

<!-- 
    Please provide the following information: 

    Description of the the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->

<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->
@0xc0170 @hugueskamba 
<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

hugueskamba committed:
CMake: remove version number from ARM toolchain

The version number was required previously when ARM Compiler 5
and ARM Compiler 6 were both supported. There was a significant
underlying change between the two that justified having distinct
build options settinfgs. It is very unlikely that such a major change will happen
between ARM Compiler 6 and future versions. It is therefore also very unlikely
that distinct option settings for future versions will be needed.
We should therefore avoid appending version number the same way we do not
append it to GCC toolchain configuration settings.

hugueskamba committed:
CMake: Add design document

hugueskamba committed:
CMake: Intentionally untrack CMake generated files

hugueskamba committed:
CMake: Add CMake scripts

hugueskamba committed:
CMake: Add support for cmsis dir

hugueskamba committed:
CMake: Add support for components dir

hugueskamba committed:
CMake: Add support for connectivity dir

hugueskamba committed:
CMake: Add support for drivers dir

hugueskamba committed:
CMake: Add support for events dir

hugueskamba committed:
CMake: Add support for features dir

hugueskamba committed:
CMake: Add support for hal dir

hugueskamba committed:
CMake: Add support for platform dir

hugueskamba committed:
CMake: Add support for rtos dir

hugueskamba committed:
CMake: Add support for storage dir

hugueskamba committed:
CMake: Add support for K64F target

hugueskamba committed:
CMake: Add support for NRF52840_DK target

hugueskamba committed:
CMake: Add support for DISCO_L475VG_IOT01A target

hugueskamba committed:
CMake: Add support for WIO_3G target

hugueskamba committed:
CMake: Add Travis script to run a basic CI check for PRs

0xc0170 committed:
CMake design doc: update to the latest mbed-tools updates

Refactor Mbed OS CMake scripts section - details about mbed-os/cmake scripts
Add mbed-tools - how mbed-tools built an application with CMake

urutva committed:
cmake: Stringify develop.cmake

Cleanup develop.cmake to remove ARMC5 and uArm compiler flags and
stringify cmake variables.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

urutva committed:
cmake: Using Arm compiler from Mbed Studio

Added a command line option `-DMBEDIDE=ON` which allows using Arm
compiler shipped with Mbed Studio on local machines.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

hugueskamba committed:
CMake: Fix failure due to restructure of rtos and ppp directories

0xc0170 committed:
Travis: add more examples we should support

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

hugueskamba committed:
CMake: Fix CI error due to dir restruct of rtos and storage dirs

urutva committed:
cmake: Re-order cmake script inclusions

The script `cmake/env.cmake` enables languages and in the case of Arm
compiler, `CMAKE_SYSTEM_PROCESSOR` is expected to be set before a
language is enabled. `CMAKE_SYSTEM_PROCESSOR` is set in
`cmake/core.cmake`. Therefore, `cmake/core.cmake` is included before
`cmake/env.cmake`.

Similarly, `cmake/profile.cmake` sets compiler options, hence it is
included before enabling languages in `cmake/env.cmake`.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

urutva committed:
cmake: Fix arm compiler build failures

Update arm compiler flags which were causing build failures.

Remove `-MMD` flag as it is not recognised by the Arm compiler.

There are no common flags between compiler, assembler and linker. The
compiler expects `-mcpu` whereas assembler and linker expects `--cpu`.

The `COMMON_FLAGS` in `cmake/toolchains/ARM.cmake` are not recognised by
assembler and linker.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

0xc0170 committed:
CMake env: fix copyright

0xc0170 committed:
CMake env: remove components, needs design first

0xc0170 committed:
CMake: fix missing copyright

0xc0170 committed:
CMake util: fix copyright

Patater committed:
cmake: Remove IAR support

IAR isn't supported anymore. Remove references to IAR from the CMake
build system generator scripts.

urutva committed:
cmake: Armclang - Fix assembler and linker flags

For Armclang, change compiler flag to `-mcpu` and assembler and linker
flag to `--cpu`.

Signed-off-by: Devaraj Ranganna <devaraj.ranganna@arm.com>

hugueskamba committed:
CMake: Use modern CMake techniques to set toolchain options

Use target_compile_options(), target_compile_definitions(),
and target_link_options() to set toolchain options.

hugueskamba committed:
CMake: Remove unsupported ARM toolchain support

hugueskamba committed:
CMake: Fix failure due to kvstore and cellular dirs changes

0xc0170 committed:
CMake: using CMAKE_BUILD_TYPE for choosing a profile

Select a profile using `-DCMAKE_BUILD_TYPE=<profile>`.
We support 3 profiles: Debug, Develop and Release.

hugueskamba committed:
CMake: Ensure targets creation before usage in `target_` functions

Move target creation statements before include statements. The targets
must be created before they can be used in `target_` functions.

hugueskamba committed:
CMake: Issue a fatal error if a lower version of CMake is used

rwalton-arm committed:
Remove gen_config interface library target

hugueskamba committed:
CMake: Fix build due to system_storage and netsocket dir move in master

0xc0170 committed:
CMake build type: set default Develop profile

Adding also comments for multiconfig/property set.

0xc0170 committed:
CMake profile: fix alignment

0xc0170 committed:
Travis: use build type for cmake

0xc0170 committed:
CMake profile: fix comments

rwalton-arm committed:
Add tools generated definitions and properties

0xc0170 committed:
Travis cmake: add testing all profiles

0xc0170 committed:
CMake toolchain: set optimization in profiles

0xc0170 committed:
CMake: set debug optimizations

0xc0170 committed:
CMake: set develop optimizations

0xc0170 committed:
CMake: set release optimizations

0xc0170 committed:
CMake toolchain: clear flags to be able to overwrite them

As Mbed uses different optimizations levels, we should be able to overwrite this via toolchain file.

0xc0170 committed:
CMake GCC ARM: use nosys specs for linker

Use retarged version (nosys.specs) for Mbed OS. To resolve an issue with debug/develop profiles.
Otherwise we get kill or getpid errors from linker, as we use nosys lib but no specs defined.

0xc0170 committed:
CMake: move env to toolchain file (#13369)

Toolchain sets target toolchain related settings that were in env. We do not need env cmake.
See https://cmake.org/cmake/help/v3.8/manual/cmake-toolchains.7.html for details

hugueskamba committed:
CMake: Fix failure due to events and netsocket dirs changes

hugueskamba committed:
CMake: Remove app.cmake module

This module was doing things that mbed-os and the application were supposed to do.
Moved the statements form the module to `mbed-os/CMakeLists.txt` and to
`<APPLICATION_ROOT>/CMakeLists.txt` .
Mbed OS also ensures the executable produced uses whatever name the application has set.

0xc0170 committed:
CMake GCC ARM: add -g3 to the common

Remove -g as g3 already provided

hugueskamba committed:
CMake: Remove add_executable

Instead provide functions to set the pre and post link commands.

hugueskamba committed:
CMake: Fix build failure due to nanostack drivers move

0xc0170 committed:
CMake toolchain: fix options comment

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

0xc0170 committed:
CMake: Profile comments update

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

hugueskamba committed:
CMake: Break down RTX CMake source file into 2 source files

hugueskamba committed:
CMake: Fix ARM toolchain compilation

Ensure the MCU core CMake module is added before the toolchain
CMake module as the latter requires CMAKE_SYSTEM_PROCESSOR
to be set when configuring the ARM toolchain.

hugueskamba committed:
CMake: Move MBED_STUDIO_ARM_COMPILER to ARM.cmake module

As its name implies, the flag is specific to the ARM toolchain.

hugueskamba committed:
CMake: Remove MCU core specific option from ARM ASM options

hugueskamba committed:
CMake: Fix detection of RTX assembly file

Access MBED_TARGET_LABELS as a global variable instead of a property.
At this point the evaluation of the property MBED_TARGET_LABELS
created in the top level CMakeLists.txt results in an empty string.

0xc0170 committed:
Update cmake/toolchain.cmake

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

hugueskamba committed:
CMake: Do not hard code target in CPU core CMake module

Provide a function in each CPU core module to set toolchain options.
All the functions have the same interface therefore the CMake source
file including the module always calls the same function regardless
of which cpu core module is actually included.
When the `mbed-os` target is broken up into multiple libraries, a
library other than `mbed-os` might need to set the toolchain options.
This will be possible by simply including the module and calling the
function with the target and toolchain for which to set the options.

hugueskamba committed:
CMake: Remove toolchain module dependency on cpu core CMake modules

Set `CMAKE_SYSTEM_PROCESSOR` in toolchain.cmake based on
`MBED_CPU_CORE`.
This information is needed by the ARM toolchain (and possibly other
toolchains), it therefore makes sense to set in `toolchain.cmake`.

hugueskamba committed:
CMake: Do not hard code target in toolchain CMake module

Provide a function in each toolchain module to set the toolchain
specific options. All the functions have the same interface
therefore the CMake source file including the module always calls
the same function regardless of which toolchain module is actually
included.
When the `mbed-os` target is broken up into multiple libraries, a
library other than `mbed-os` might need to set the toolchain options.
This will be possible by simply including the module and calling the
function with the target for which to set the options.

hugueskamba committed:
CMake: Do not hard code target in build profile CMake module

Provide a function in each build profile module to set the toolchain
options. All the functions have the same interface therefore the
CMake source file including the module always calls the same
function regardless of which build profile module is actually
included.
When the `mbed-os` target is broken up into multiple libraries,
a library other than `mbed-os` might need to set the toolchain
options. This will be possible by simply including the module
and calling the function with the target and toolchain for which
to set the options.

hugueskamba committed:
CMake: Remove utility function dependency on mbed-os target

The utility function may be called by targets other than mbed-os
when it is broken into multiple libraries.

hugueskamba committed:
CMake: Fix build failure due to drivers and lwipstack dirs move

hugueskamba committed:
CMake: Fix build issue due to platform changes

hugueskamba committed:
CMake: Fix access to cellular drivers

hugueskamba committed:
CMake: Fix access to usb hal source file

hugueskamba committed:
CMake: Fix WIO_3G scatter file shebang to work with ARM Compiler 6

hugueskamba committed:
CMake: Minor formatting of top level CMakeLists.txt source file

hugueskamba committed:
CMake: Fix build failure due to nanostack dir move

hugueskamba committed:
CMake: Set language std per target using properties

hugueskamba committed:
CMake: Remove default build type selection from top level source file

hugueskamba committed:
CMake: Fix breakage due to BLE dir move

hugueskamba committed:
CMake: Fix failure due to BLE component dir move

hugueskamba committed:
CMake: Use `mbed_` prefix for CMake functions

hugueskamba committed:
CMake: Fix failure due to MbedTLS rdir relocation

rwalton-arm committed:
CMake: Check CMake builds CMake-capable targets in Travis

0xc0170 committed:
CMake: use response files

As Mbed OS is built as whole, we have long paths again. This is known issue with windows. To fix the paths, we need to use response files.

Ninja is special, needs to be forced to use long paths.

hugueskamba committed:
CMake: Fix build failure due to lora, cryptocell, and nfc dirs move

0xc0170 committed:
CMake: Fix include path for Cryptocell

hugueskamba committed:
CMake: Fix NRF52840_DK build failure

* Remove duplicated ARM toolchain directory
* Link with archived cryptocell 310 libraries instead
  of adding them as source files
* Provide default MBED_BOOT_STACK_SIZE definition
  in scatter file as is done for GCC_ARM linker file

hugueskamba committed:
CMake: Fix CORDIO_LL build failure

List missing CORDIO_LL source file in CMake source file.

hugueskamba committed:
CMake: Fix failures due to device_key and storage dirs move

0xc0170 committed:
cmake: fix sd component recent dir structure

0xc0170 committed:
CMake: use CMAKE_TOOLCHAIN_FILE

We should define CMAKE_TOOLCHAIN_FILE (as we are providing Mbed OS toolchain file). This fixes
also the issue we had with Unix makefiles where application would be without response files.

For more details, see https://gitlab.kitware.com/cmake/cmake/-/issues/21104

As the Mbed OS toolchain file is included by CMake, we need to have enabling language in the main CMake (can be executed
only once).

0xc0170 committed:
CMake: Fix blockdevice components source and include paths

This file paths were updated on master and this was broken after
rebasing.

hugueskamba committed:
CMake: Fix build failure due to PSA dir move

hugueskamba committed:
CMake: List missing files in sal-stack-nanostack

hugueskamba committed:
CMake: Add support for K66F

hugueskamba committed:
CMake: Correct file listing for PSA

hugueskamba committed:
CMake: Fix selection of scatter file and startup file for DISCO_L475_IOT01A

0xc0170 committed:
CMake ARMClang6: use armclang for asm

Fixes the issue of armasm, passing -DMACRO={0,0} - fails to parse.

We use armclang frontend but passing -masm=armasm.

0xc0170 committed:
CMake cores: fix flag  mpcu for asm

hugueskamba committed:
CMake: Fix build failure due to CMSIS restructure

0xc0170 committed:
CMake ARMClang: use -masm=auto

Auto mode for ASM files. There might be rare cases where this fails, we will fix
them by requesting the component to use gnu syntax for assembly files.

hugueskamba committed:
CMake: Fix build due to hal dir move

0xc0170 committed:
CMake: 3.18.2 minimum version set

There are fixes in the latest version we need for ARMClang.

hugueskamba committed:
CMake: mbed_add_cmake_directory_if_labels checks presence of CMakeLists.txt before adding subdirectory

0xc0170 committed:
CMake: Add information about the current state of support (#13551)

hugueskamba committed:
CMake: Fix build failure due to mbed-trace, mbed-client-randlib, and nanostack dir changes

hugueskamba committed:
CMake: Clean up platform

0xc0170 committed:
CMake: Fix GCC_ARM pre-linking stage (#13575)

Make it generic for all targets. It was previously hardcoded for K64F

hugueskamba committed:
CMake: Revert mbed-trace CMake input source file to match upstream repository (#13590)

As there is currently no mbed-trace library, statements related to that
target have been commented out.

hugueskamba committed:
CMake: Remove TOOLCHAIN_ARM_STD dir for TARGET_STM32F439xI

hugueskamba committed:
CMake: Fix build failure due to BLE dir changes

hugueskamba committed:
CMake: Fix ARM toolchain configuration

mbed_generate_gcc_options_for_linker is to
be called only when GCC_ARM is selected.

hugueskamba committed:
CMake: Fix DISCO_L475VG_IOT01A ARM toolchain support

Correct location of startup code and linker file.

hugueskamba committed:
CMake: Add background and future plans to README (#13589)

hugueskamba committed:
CMake: Move modules under tools subdirectory

hugueskamba committed:
CMake: Add support for printf lib selection

hugueskamba committed:
CMake: Add support for C lib selection

Configure the selected toolchain for the selected C library

0xc0170 committed:
CMake design: update to the latest version of feature-cmake branch (#13614)

We made changes like tools/cmake folder change, removed app.cmake.

hugueskamba committed:
CMake: Fix location of M33 exception handler source file

hugueskamba committed:
CMake: Fix TFM targets build

Add linking time pre-processor macro DOMAIN_NS
for TFM targets and TrustZone non secure targets.

hugueskamba committed:
CMake: Add support for NUCLEO_F401RE

rajkan01 committed:
CMake: Add a function to generate executable artifacts with memap (#13695)

- Add linker option to generate map file for GCC_ARM and ARM toolchain
- Add mbed_generate_map_file function to call memap.py (parse the map
file).
- Add mbed_generate_executable function to generate executable artifacts with a memory map table

hugueskamba committed:
CMake: Componentize Mbed OS into multiple CMake targets (#13732)

Aside from the core mbed-os CMake target, a number of targets have been created so they can optionally be included by application executables that require them using `target_link_libraries()`.

Co-authored-by: Martin Kojtal <martin.kojtal@arm.com>
Co-authored-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@arm.com>

rajkan01 committed:
CMake: replace usage of the mbed_add_cmake_directory_if_labels() function (#13754)

Directories that start with special prefixes (TARGET_, FEATURE_, COMPONENT_)  are added to the build based on Mbed target configuration from targets.json instead of calling utility function mbed_add_cmake_directory_if_labels().

0xc0170 committed:
cmake: fix present inclusion

We use only _PRESENT macros for components if they are enabled. Use the same in
the mbed.h file or anywhere else.

0xc0170 committed:
CMake: fix for Gcc Arm preprocessing linker file

A linker script needs symbols (stack size, app size, etc). They are basic values or defines. Not any array like or string like macros. We should filter these, as they are not valid anyway.

The other option to fix this would be to fix all the macros but I dont think it is needed as these config values won't be used in the linker script anyway.

Not allowed in ld files macros with spaces, like MACRO={0, 2, 3} or MACRO=(4 * 2000).

hugueskamba committed:
CMake: Replace tested BLE application in CI description

BLE_Button, which was the previously tested BLE example, is
being removed in mbed-os-example-ble. Replace it with BLE_LED.

0xc0170 committed:
CMake: Fix travis fetch PR (#13763)

Either fetch TRAVIS_COMMIT or the entire pull request (depending on the TRAVIS_PULL_REQUEST environment variable).

hugueskamba committed:
CMake: Fix BLE private address linker error

List newly added files in CMake input source files.

rajkan01 committed:
CMake: split rtos from core target

Add or remove and directories containing source files, header files, and macro definitions that implement RTOS support.
This includes:
- cmsis/CMSIS_5/CMSIS/RTOS2/
- cmsis/device/rtos/
- rtos/source/ConditionVariable.cpp
- rtos/source/Thread.cpp

0xc0170 committed:
CMake: remove string.h in STM32 usb, already included (#13773)

hugueskamba committed:
CMake: Restore rtos api and netsocket presence macro names

The old tools rely on the old names to build

hugueskamba commented on CMakeLists.txt:
```diff
@@ -99,6 +99,13 @@ add_subdirectory(drivers/source/usb EXCLUDE_FROM_ALL)
 add_subdirectory(features EXCLUDE_FROM_ALL)
 add_subdirectory(platform/FEATURE_EXPERIMENTAL_API EXCLUDE_FROM_ALL)
 
+if(${MBED_BAREMETAL_PROFILE})
```
I think it is more likely that a board does not support the full profile than it does not support the baremetal profile.
In any case, you should check that the rtos profile is supported if `MBED_BAREMETAL_PROFILE` is false and also that the baremetal profile is supported if `MBED_BAREMETAL_PROFILE` is true.

Also, please add a comment above this, something like:
`Verify that the selected application OS profile is supported by the Mbed target`

rajkan01 committed:
Update CMake based on new STM32L4 directory structure

0xc0170 committed:
CMake: fix application config

These settings like enable_language should be done in the application and just once.
We hit the issue when you expose sources to an app (interface or public), CMake errors as some of the internal settings have not been configured.

0xc0170 committed:
CMake: remove TODO as it is not valid anymore

hugueskamba committed:
CMake: Rename CMake targets

* mbed-os renamed mbed-core
* mbed-os-<COMPONENT> renamed mbed-<COMPONENT>

hugueskamba committed:
CMake: Create a mbed-os alias library

This ensures that applications which link with the mbed-os
CMake target still work. We want applications that require the OS
to use the mbed-os CMake target as we will be creating a new one
that will include mbed-core and mbed-rtos.

rajkan01 committed:
CMake: add mbed-os and mbed-baremetal targets

mbed-os consists of mbed-core and mbed-rtos
mbed-baremetal consists of mbed-core

The main change is for mbed-core. Changing from object library to be interface. This way it allows us to do the above to have 2 main targets for users to use.

This should be backward compatible change as mbed-os target we used contains the same files/options as previously set.

0xc0170 committed:
Update rtos/CMakeLists.txt

Co-authored-by: Hugues Kamba <41612201+hugueskamba@users.noreply.github.com>

hugueskamba committed:
CMake: Add support for NUCLEO_F303K8

Added the board so it can be used for testing the bare-metal profile on a Mbed 2 board

hugueskamba committed:
Merge pull request #17 from hugueskamba/hk_feature-cmake-rtos-lib

CMake: Add support for NUCLEO_F303K8

0xc0170 committed:
Merge pull request #13818 from 0xc0170/feature-cmake-rtos-lib

CMake: Adding mbed-core and mbed-baremetal targets

rajkan01 committed:
Replace PUBLIC to INTERFACE keyword

rajkan01 committed:
CMake: Add small library selection by default for baremetal profile

0xc0170 commented on CMakeLists.txt:
```diff
@@ -94,6 +74,38 @@ add_subdirectory(platform)
 add_subdirectory(targets)
 
+#Verify that the selected application OS profile is supported by the Mbed target
+if(NOT MBED_CONF_RTOS_PRESENT)
```
`MBED_CONF_RTOS_PRESENT` is define, not CMake variable - does this work?

hugueskamba replied:
@rajkan01 How was this tested?

rajkan01 replied:
It works

hugueskamba replied:
How was it tested?

0xc0170 commented on CMakeLists.txt:
```diff
@@ -94,6 +74,38 @@ add_subdirectory(platform)
+            "Baremetal profile is not supported for this target")
+    else()
+        set(MBED_C_LIB "small" CACHE STRING "")
+        message(${MBED_C_LIB})
```
is this debug print,not needed? or if we want to say something, it should be more detailed.

0xc0170 commented on CMakeLists.txt:
```diff
@@ -151,18 +163,18 @@ function(mbed_generate_bin_hex target)
-            COMMAND ${CMAKE_COMMAND} -E echo "-- built: $<TARGET_FILE:${target}>.bin"
-            COMMAND ${elf_to_bin} -O ihex $<TARGET_FILE:${target}> $<TARGET_FILE:${target}>.hex
-            COMMAND ${CMAKE_COMMAND} -E echo "-- built: $<TARGET_FILE:${target}>.hex"
+            COMMAND ${elf_to_bin} -O binary $<TARGET_FILE:${target}> ${CMAKE_BINARY_DIR}/${target}.bin
```
this is another PR, should not be in this one?

0xc0170 replied:
I would remove the dependency here, not needed in this Pr

0xc0170 commented on cmsis/device/rtos/CMakeLists.txt:
```diff
@@ -35,3 +35,14 @@ target_compile_definitions(mbed-rtos
+
+if(NOT "full" IN_LIST MBED_TARGET_SUPPORTED_APPLICATION_PROFILES)
+   message(FATAL_ERROR
+       "Full profile is not supported for this target")
```
What does this mean,  we should use baremetal instead? If yes, we might want to add it here (for a new user it might be helpful hint).
